app/components/modal/restaurant/restaurant.modal.controller.js

```diff
@@ -0,0 +1,73 @@
+(function () {
+    'use strict';
+    angular.module('FoodOrderingApp')
+        .controller('RestaurantModalController', RestaurantModalController);
+
+    RestaurantModalController.$inject = ['$uibModalInstance', '$log', 'RestaurantService', 'APP_CONSTANT', 'delRestaurant', 'editRestaurant'];
+
+    function RestaurantModalController($uibModalInstance, $log, RestaurantService, APP_CONSTANT, delRestaurant, editRestaurant) {
+        var vm = this;
+
+        vm.restaurant = {};
+        vm.resNameReqMsg = APP_CONSTANT.RES_NAME_REQ_MSG;
+        vm.contactNoMsg = APP_CONSTANT.CONTACT_NO_MSG;
+        vm.numbersOnlyMsg = APP_CONSTANT.NUMBERS_ONLY_MSG;
+
+
+        init();
+
+        function init() {
+            if (editRestaurant) {
+                vm.restaurant = angular.copy(editRestaurant);
+                vm.copyRestaurant = angular.copy(editRestaurant);
+            }
+
+            if (delRestaurant) {
+                vm.restaurantToBeDeleted = delRestaurant.name;
+            }
+        }
+
+        vm.addRestaurant = function (restaurant) {
+            RestaurantService.addRestaurant(restaurant)
+                .then(
+                    function (answer) {
+                        $uibModalInstance.close(answer.data);
+                    },
+                    function (error) {
+                        $log.info(error);
+                    }
+                );
+            RestaurantService.setAlertMessage(restaurant.name + ' ' + APP_CONSTANT.ADD_MSG);
+        };
```

Comment:
RestaurantService.setAlertMessage(restaurant.name + ' ' + APP_CONSTANT.ADD_MSG); this part need do goes inside before  $uibModalInstance.close(answer.data);. Only when restaurant had been added successfully alert the message.
Same thing need to goes in below functions.